02-intermediate-ruby/object-inheritance.md

```diff
@@ -3,77 +3,138 @@
 - Describe objects in terms of _is-a_ relationships
 - Use _inheritance_ to indicate a hierarchy of ownership or lineage between classes
 - Explore overriding method definitions in _subclasses_
-- Use `super` to invoke the behavior of a _base class_ in a _subclass 
+- Use `super` to invoke the behavior of a _base class_ in a _subclass
 
 ## Object Inheritance: _is-a_
-_Inheritance_ is used to model the behaviors of one class after another. We would want to do this when two objects have a clear chain of ownership. Often the parent class, or _base class_, is more general while the child, or _subclass_, is more specific.
+_Inheritance_ is used to model the behaviors of one class after another. We would want to do this when two objects have a clear relationship. Often the parent class, or _base class_, is more general while the child, or _subclass_, is more specific.
 
-For example, we could have a `Bird` class that describes the general behaviors of birds. This is a _base class_. We could create `Penguin` and `Hawk` classes that model more specific behaviors. A `Penguin` is very different than a `Hawk`, but they are both birds. It would make sense then, to declare them _subclasses_ of `Bird`. 
+For example, we could have a `Product` class that describes the general behaviors of items in an online store. This is a _base class_ or _superclass_. We could create `Book` and `Video` classes that model more specific behaviors and additional information. A `Book` is very different than a `Video`, but they are both products in the store. It would make sense then, to declare them _subclasses_ of `Product`.
 
-A way to describe this is by using the _is-a_ phrase: `Penguin` _is-a_ `Bird`. `Hawk` _is-a_ `Bird`. In Ruby, we use the `<` symbol to denote that the class on the left of the `<` symbol is inheriting behaviors from the class on the right side of the `<` symbol:
+A way to describe this is by using the _is-a_ phrase: a `Book` _is-a_ `Product`. `Video` _is-a_ `Product`. In Ruby, we use the `<` symbol to denote that the class on the left of the `<` symbol is inheriting behaviors from the class on the right side of the `<` symbol:
 
 ```ruby
-class Bird
-  def talk
-    "I'm a bird. Kah."
+class Product
+  attr_reader :id, :name
+
+  def initialize
+    @id = 12345
+    @name = "Generic Product"
+  end
```

Comment:
Can you make `id` and `name` be parameters for the constructor? Having them hard-coded feels really goofy.

Reply:
I can, I just didn't want to jump into using `super` in the subclass' constructor initially.